bliss/simulator/toy_simulated_dataset.py

```diff
@@ -0,0 +1,380 @@
+import dataclasses
+import math
+import warnings
+
+import torch
+from torch.utils.data import IterableDataset
+
+from bliss.catalog import FullCatalog
+
+# prevent pytorch_lightning warning for num_workers = 0 in dataloaders with IterableDataset
+warnings.filterwarnings(
+    "ignore", ".*does not have many workers which may be a bottleneck.*", UserWarning
+)
+
+
+def counts_mask(counts):
+    """Identify the first `counts[i]` elements of the ith row of the result.
+
+    Args:
+        counts: (m) tensor of counts
+
+    Returns:
+        (m,M) tensor where M = counts.max() and result[i, j] = j < counts[i]
+
+
+    Example:
+    counts = torch.tensor([2,1,1])
+    counts_mask(counts)
+    tensor([[True, True],
+            [True, False],
+            [True, False]])
+
+    """
+    # Determine the max length for the range
+    max_count = counts.max()
+
+    # Create a range tensor [0, 1, 2, ..., max_count-1]
+    range_tensor = torch.arange(max_count, device=counts.device).unsqueeze(0)
+
+    # Expand counts to match the shape for broadcasting
+    expanded_counts = counts.unsqueeze(1)
+
+    # Compute the mask
+    return range_tensor < expanded_counts
+
+
+def draw_trunc_pareto(b, c, loc, scale, size, torch_generator=None):
```

Comment:
We used to have code like this in `prior.py` but now we use ``scipy.stats.truncpareto`.
Would it be possible to have your simulator take a prior as argument, the way `SimulatedDataset` does, for greater symmetry? Presumably you'd just ignore some of the fields for now (e.g., source type). That would make this more of a drop-in replacement for `SimulatedDataset`, which I think is the goal, at least if we're including this code in our main `bliss/bliss` directory (as opposed to a case study).